### DISCUSSION TITLE
Uninstall for a noob.

### DISCUSSION THREAD

**Discussion by Stolowma** [Staubgeborener/klipper-backup#139] [General]
I just started converting my printer to Klipper with a BTT Pad 7 with the CM1 board. Got everything working and when I saw info about this I thought "great!", took too long to set this up to lose everything. It worked perfectly the first time, saved everything and I was thrilled. Unfortunately next boot up it wouldn't connect to the network and the touch screen freezes up. I flashed the stock firmware from BTT and everything worked. I installed this again and now have the same issue. I'd like to uninstall Klipper-Backup to see if this is what's causing the problem but I don't no much about Linux or programming in general. Is there an automated way to do this? Thanks

**Comment by Staubgeborener** [comment]
To be honest, I don't know how Klipper-Backup could do something like this. It is a simple shell script without binaries or similar. Klipper-Backup has few dependencies that may need to be installed (git, jq, curl, rsync) and I don't see any connection to your problem there either.

If you still want to "uninstall" (which is the wrong term, more acurate would be to "remove", as I said, no binaries are installed)  it:
`sudo rm -r ~/klipper-backup ~/config_backup`

If you have installed systemd services (backup on boot, etc), those files are stored in `/etc/systemd/system` and can be removed like `sudo rm /etc/systemd/system/[name-of-the-service]`.

You may also have generated entries in crontab: check via `sudo crontab -e` and remove the Klipper-Backup lines.

Oh and there should be a entry in moonraker.conf which you also need to remove (but this can be done with the mainsail/fluidd web ui).
Tip: The next time you test it, install it via [KIAUH](https://klipperbackup.xyz/installation/#installation). In this context, KIAUH also has a nice “remove” option which does all the things described above automatically.

**Comment by Stolowma** [comment]
Thanks, even with my limit understanding of how all this works I found it strange too. I just want to check and see if that's the problem before I return the Pad 7. I tried using KIAUH to uninstall but, I'm assuming because I didn't install it using that, it wasn't listed. If I install it again I will. Thanks for your work on this I have a feeling it will be a lifesaver.

**Comment by Staubgeborener** [comment]
> I tried using KIAUH to uninstall but, I'm assuming because I didn't install it using that, it wasn't listed. 

Correct.

**Comment by Staubgeborener** [comment]
Just in case if you are still struggling, this should delete every little fragment of Klipper-Backup (if not installed with KIAUH):

```
sudo rm -r ~/klipper-backup ~/config_backup

# see: https://klipperbackup.xyz/automation/#how-do-i-deactivate-systemd-services
sudo systemctl stop klipper-backup-on-boot.service
sudo systemctl disable klipper-backup-on-boot.service

sudo systemctl stop klipper-backup-filewatch.service
sudo systemctl disable klipper-backup-filewatch.service

sudo rm /etc/systemd/system/klipper-backup-*
```

Two things left:
1. Remove the [moonraker entry](https://klipperbackup.xyz/updating/#moonraker-update-manager)
2. Remove the [cronjob](https://klipperbackup.xyz/automation/#cron) if you have used this (`crontab -e`, delete Klipper-backup line, `STRG+X` -> `Y`)

**Reply by Stolowma** [reply]
Thanks, I'll give it a try and let you know if that was causing the issue.
